File: rotateString.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        int n=s.size();
        while(n--){
            char ch = s[0];
            for(int i=1;i<s.size();i++){
                s[i-1]=s[i];
            }
            s[s.size()-1]=ch;
            if(s==goal) return true;
        }
        return false;
    }
};

File: rotateString.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length()!=goal.length()){
            return false;
        }
        int freq[] = new int[26];
        int n = s.length();
        for(int i=0;i<n;i++){
            freq[s.charAt(i)-'a']++;
            freq[goal.charAt(i)-'a']--;
        }
        for(int count=0;count<freq.length;count++){
            if(freq[count]!=0){
                return false;
            }
        }
        return true;
    }
}
